src/Contexts/Adaptors/EVMAdaptors/EVMHomeAdaptorProvider.tsx

```diff
@@ -131,83 +198,66 @@ export const EVMHomeAdaptorProvider = ({
       const supported = !!chainbridgeConfig.chains.find(chain => chain.networkId === network);
       setNetworkSupported(supported);
       if (!!network && !supported) {
-        resetOnboard();
+        disconnectWallet();
         setWalletType("unset");
       }
       if (chain) {
+        // `setTokens` maps over the chain tokens. For each token, it checks if it exists in the wallet.
+        // If it does, it uses the wallet's balance. If it doesn't, it sets balance as "0".
+        // The function constructs an object where each key is a token address and its value is the token details and balance.
+        setTokens(chain.tokens.reduce((acc, currentToken) => {
+          const matchingTokenInWallet = wallet?.accounts[0]?.secondaryTokens?.find(
+              (walletToken) => walletToken.name === currentToken.symbol
+          );
+
+          // Calculate balance by converting from 'wei' to 'ether' and then to the token's units
+          // If `matchingTokenInWallet` does not exist, set balance to "0"
+          const balance = matchingTokenInWallet
+              ? +matchingTokenInWallet.balance * Math.pow(10, 18) / Math.pow(10, currentToken.decimals || 10)
```

Comment:
Why `|| 10` can happen? Why do we assume that this number should be 10 for any chain?

Reply:
Basically, we use only Cere tokens for Ethereum chains with decimals equal to 10. I have decided to add it as a placeholder, but I don't think it's the most elegant solution.